File: 2-add-two-numbers/2-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode(-1)
        dummy = head
        if not l1: return l2
        if not l2: return l1
        carry = 0
        while(l1 or l2):
            firstval = l1.val if l1 else 0
            secondval = l2.val if l2 else 0
            total = firstval + secondval + carry
            carry = total//10
            total = total%10
            newnode = ListNode(total)
            dummy.next = newnode
            dummy = dummy.next
            l1 = l1.next if l1 else l1
            l2 = l2.next if l2 else l2
        if carry: dummy.next = ListNode(1)
        return head.next

File: 2-add-two-numbers/2-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummyNode;  ListNode* head;
        dummyNode = head = new ListNode(-1);
        if(!l1) return l2;
        if(!l2) return l1;
        int carry = 0;
        while(l1||l2){
            int first = l1?l1->val:0;
            int second = l2?l2->val:0;
            int total = first+second+carry;
            carry = total/10;
            total %= 10;
            ListNode* newNode= new ListNode(total);
            dummyNode->next = newNode;
            dummyNode = dummyNode->next;
            l1 = l1?l1->next:l1;
            l2 = l2?l2->next:l2;
        }
        if(carry)
            dummyNode->next = new ListNode(1);
        return head->next;
    }
};